src/lua/lua_http.c

```diff
@@ -316,16 +316,31 @@ static void
 lua_http_push_headers (lua_State *L, struct rspamd_http_message *msg)
 {
 	const char *name, *value;
+	gint i;
 
 	lua_pushnil (L);
 	while (lua_next (L, -2) != 0) {
 
 		lua_pushvalue (L, -2);
 		name = lua_tostring (L, -1);
-		value = lua_tostring (L, -2);
-
-		if (name != NULL && value != NULL) {
-			rspamd_http_message_add_header (msg, name, value);
+		if (lua_istable (L, -2)) {
```

Comment:
I'd suggest using of `rspamd_lua_table_size` instead of this heuristic.

Reply:
Do you mean instead of `for (i = 1; ; i++) { .. if (lua_isnil (L, -1)) {` story? I'm not sure what to change on this line.

Reply:
If you check code of the function (rspamd_lua_table_size) it returns 0 for a non-table.

Reply:
Yes, I saw this, but I was worried about case of empty table. :)

Reply:
It should be treated as an error I suppose. `set_header('name', '')` or `set_header('name', {''})` should be absolutely fine on the other hand.